admin_manual/installation/nginx.rst

```diff
@@ -114,75 +76,109 @@ webroot of your nginx installation. In this example it is
-          deny all;
+      
+      # HTTP response headers borrowed from NextCloud `.htaccess`
+      add_header Referrer-Policy                      "no-referrer"   always;
+      add_header X-Content-Type-Options               "nosniff"       always;
+      add_header X-Download-Options                   "noopen"        always;
+      add_header X-Frame-Options                      "SAMEORIGIN"    always;
+      add_header X-Permitted-Cross-Domain-Policies    "none"          always;
+      add_header X-Robots-Tag                         "none"          always;
+      add_header X-XSS-Protection                     "1; mode=block" always;
+      
+      # Remove X-Powered-By, which is an information leak
+      fastcgi_hide_header X-Powered-By;
+      
+      # Path to the root of your installation
+      root /var/www/nextcloud;
+      
+      # Specify how to handle directories -- specifying `/index.php$request_uri`
+      # here as the fallback means that Nginx always exhibits the desired behaviour
+      # when a client requests a path that corresponds to a directory that exists
+      # on the server. In particular, if that directory contains an index.php file,
+      # that file is correctly served; if it doesn't, then the request is passed to
+      # the front-end controller. This consistent behaviour means that we don't need
+      # to specify custom rules for certain paths (e.g. images and other assets,
+      # `/updater`, `/ocm-provider`, `/ocs-provider`), and thus
+      # `try_files $uri $uri/ /index.php$request_uri`
+      # always provides the desired behaviour.
+      index index.php index.html index.htm /index.php$request_uri;
+      
+      # Default Cache-Control policy
+      expires 1m;
+      
+      # Rule borrowed from `.htaccess` to handle Microsoft DAV clients
+      if ( $http_user_agent ~ DavClnt ) { rewrite ^/$ /remote.php/webdav/ redirect; }
+      
+      location = /robots.txt {
+          allow all;
+          log_not_found off;
+          access_log off;
       }
-      location ~ ^\/(?:\.|autotest|occ|issue|indie|db_|console) {
-          deny all;
+      
+      # Make a regex exception for `/.well-known` so that clients can still
+      # access it despite the existence of the regex rule `location ~ /\.`
+      # which would otherwise handle requests for `/.well-known`.
+      location ^~ /.well-known {
+          # The following 6 rules are borrowed from `.htaccess`
+      
+          rewrite ^/\.well-known/host-meta\.json  /public.php?service=host-meta-json  last;
+          rewrite ^/\.well-known/host-meta        /public.php?service=host-meta       last;
+          rewrite ^/\.well-known/webfinger        /public.php?service=webfinger       last;
+          rewrite ^/\.well-known/nodeinfo         /public.php?service=nodeinfo        last;
+          
+          location /.well-known/carddav   { return 301 /remote.php/dav/; }
+          location /.well-known/caldav    { return 301 /remote.php/dav/; }
```

Comment:
The Apache regex rules only seek to match the beginning of the URI, e.g. `^/\.well-known/carddav/`, so they match URIs such as `/.well-known/carddav/whatever`

Reply:
In fact, perhaps the `.htaccess` rules need to be altered, because they also match URIs such as `/.well-known/carddavfoo`. Of course, if we know that DAV-compliant clients only ever make requests for `/.well-known/[card|cal]dav`, and not any sub-paths of these, then `location =` would be preferred.

Also, on the topic of `.htaccess` itself potentially needing to be altered, it currently reads:

```
  RewriteRule ^\.well-known/host-meta /public.php?service=host-meta [QSA,L]
  RewriteRule ^\.well-known/host-meta\.json /public.php?service=host-meta-json [QSA,L]
```

But in this order, a request for `/.well-known/host-meta.json` matches the first regex, so a redirect to `/public.php?service=host-meta` occurs rather than to `/public.php?service=host-meta-json`. Incidentally, this is why I have written the corresponding `rewrite` rules in my `location ^~ /.well-known` block in the opposite order.

Reply:
In this case I think what we should do is read up on the RFC6764 and just verify what URLs NC should be responding too. If `.htaccess` isn't doing it correctly ideally we should fix it there not try to match their quirks.

Feel free to do some reading here, or I can try looking at sometime soon.

Reply:
Had a read over the RFC, and, in conjunction with [RFC5785 §5.1](https://tools.ietf.org/html/rfc5785#section-5.1), it registers the URI suffixes `caldav` and `carddav` ([see §9.1](https://tools.ietf.org/html/rfc6764#section-9.1)). This means that the URIs `/.well-known/caldav` and `/.well-known/carddav`, as well as sub-URIs (e.g. `/.well-known/caldav/foo`, but not `/.well-known/caldavfoo`), are reserved for use by CalDAV/CardDAV. This is also reflected in [IANA's list of well-known URIs](https://www.iana.org/assignments/well-known-uris/well-known-uris.xhtml).

However, only usage of the exact URIs `/.well-known/caldav` and `/.well-known/carddav — but not any sub-URIs such as `/.well-known/caldav/foo` — appears to be defined ([see §5](https://tools.ietf.org/html/rfc6764#section-5)). As such, it seems that the Apache rules ought to be as follows:

```
  RewriteRule ^\.well-known/carddav$ /remote.php/dav/ [R=301,L]
  RewriteRule ^\.well-known/caldav$ /remote.php/dav/ [R=301,L]
```

Or at the very least, so as not to match e.g. `/.well-known/caldavfoo`:

```
  RewriteRule ^\.well-known/carddav($|/) /remote.php/dav/ [R=301,L]
  RewriteRule ^\.well-known/caldav($|/) /remote.php/dav/ [R=301,L]
```

Thus, I think we should use `location =`.

Reply:
Thanks for taking a read through! If you want to make an PR for the **.htaccess** for this/ or any other changed you feel are needed feel free to CC me and I can back you up.

I'll go ahead and implement the `location =` and resolve this comment.